src/MainWindow.vala

```diff
@@ -120,4 +124,66 @@ public class Dock.MainWindow : Gtk.ApplicationWindow {
             return false;
         });
     }
+
+    public void remove_launcher (Launcher launcher) {
+        box.remove (launcher);
+
+        if (launcher.pinned) {
+            var old_pinned_ids = settings.get_strv ("launchers");
+            string[] new_pinned_ids = {};
+
+            var to_remove_id = launcher.app_info.get_id ();
+            foreach (string app_id in old_pinned_ids) {
+                if (app_id != to_remove_id) {
+                    new_pinned_ids += app_id;
+                }
+            }
+
+            settings.set_strv ("launchers", new_pinned_ids);
+        }
+    }
+
+    public void move_launcher_after (Launcher source, Launcher? target) {
+        var before_source = source.get_prev_sibling ();
+
+        box.reorder_child_after (source, target);
+
+        string[] new_pinned_ids = {};
+        bool add = false;
+        Gtk.DirectionType dir = UP;
```

Comment:
I guess `UP` here is used as an invalid value, right? Can you add a comment about this please.